external-crates/move/move-compiler/src/diagnostics/mod.rs

```diff
@@ -374,7 +398,11 @@ macro_rules! diag {
 
 impl WarningFilters {
     pub fn new() -> Self {
-        Self(BTreeMap::new())
+        Self(BTreeMap::new(), false)
+    }
+
+    pub fn new_for_lib_code() -> Self {
+        Self(BTreeMap::new(), true)
```

Comment:
Still not entirely clear on the intent here

Reply:
I hope this one has been taken care of by explanation in https://github.com/MystenLabs/sui/pull/13800#discussion_r1327598619